pipelines/release-to-github-signed/release-to-github-signed.yaml

```diff
@@ -0,0 +1,348 @@
+            TARGET_NAMESPACE=$(echo $(params.releaseplanadmission) | cut -f1 -d/)
+
+            RELEASE_NAME=$(echo $(params.release) | cut -f2 -d/)
+            RELEASEPLAN_NAME=$(echo $(params.releaseplan) | cut -f2 -d/)
+            RELEASEPLANADMISSION_NAME=$(echo $(params.releaseplanadmission) | cut -f2 -d/)
+            SNAPSHOT_NAME=$(echo $(params.snapshot) | cut -f2 -d/)
+
+            CAN_I_READ_RELEASES=$(kubectl auth can-i get release/${RELEASE_NAME} -n ${ORIGIN_NAMESPACE})
+            CAN_I_READ_RELEASEPLANS=$(kubectl auth can-i get releaseplan/${RELEASEPLAN_NAME} -n ${ORIGIN_NAMESPACE})
+            CAN_I_READ_RELEASEPLANADMISSIONS=$(kubectl auth can-i get releaseplanadmission/${RELEASEPLANADMISSION_NAME}\
+                -n ${TARGET_NAMESPACE})
+            CAN_I_READ_SNAPSHOTS=$(kubectl auth can-i get snapshot/${SNAPSHOT_NAME} -n ${ORIGIN_NAMESPACE})
+            CAN_I_CREATE_INTERNALREQUESTS=$(kubectl auth can-i create internalrequest -n ${TARGET_NAMESPACE})
+
+            echo ""
+            echo "CAN_I_READ_RELEASES? ${CAN_I_READ_RELEASES}"
+            echo "CAN_I_READ_RELEASEPLANS? ${CAN_I_READ_RELEASEPLANS}"
+            echo "CAN_I_READ_RELEASEPLANADMISSIONS? ${CAN_I_READ_RELEASEPLANADMISSIONS}"
+            echo "CAN_I_READ_SNAPSHOTS? ${CAN_I_READ_SNAPSHOTS}"
+            echo ""
+            echo "CAN_I_CREATE_INTERNALREQUESTS? ${CAN_I_CREATE_INTERNALREQUESTS}"
+            echo ""
+
+            if [ "${CAN_I_READ_RELEASES}" == "no" ] ||\
+                [ "${CAN_I_READ_RELEASEPLANS}" == "no" ] ||\
+                [ "${CAN_I_READ_RELEASEPLANADMISSIONS}" == "no" ] ||\
+                [ "${CAN_I_READ_SNAPSHOTS}" == "no" ] ||\
+                [ "${CAN_I_CREATE_INTERNALREQUESTS}" == "no" ] ; then
+              echo "Error: Cannot read or create required Release resources!"
+              echo ""
+              echo "This indicates that your workspace is not correctly setup"
+              echo "Please reach out to a workspace administrator"
+              exit 1
+            fi
+
+            echo "Access to Release resources verified"
+    - name: collect-data
+      taskRef:
+        resolver: "git"
+        params:
+          - name: url
+            value: https://github.com/redhat-appstudio/release-service-catalog.git
+          - name: revision
+            value: main
+          - name: pathInRepo
+            value: tasks/collect-data/collect-data.yaml
+      params:
+        - name: release
+          value: $(params.release)
+        - name: releaseplan
+          value: $(params.releaseplan)
+        - name: releaseplanadmission
+          value: $(params.releaseplanadmission)
+        - name: snapshot
+          value: $(params.snapshot)
+        - name: subdirectory
+          value: $(context.pipelineRun.uid)
+      workspaces:
+        - name: data
+          workspace: release-workspace
+      runAfter:
+        - verify-access-to-resources
+    - name: validate-single-component
```

Comment:
I don't have the background context for this pipeline, but are we expecting it to only run for applications with a single component? Want to make sure the inclusion of this task wasn't just copied over and forgotten about

Reply:
For the terraform pipeline, we are only covering one component to sign and release to github. The rest of the pipeline is not compatible to releasing more than one components at the moment, so I kept this test because of that